src/libraries/Microsoft.Extensions.DependencyModel/src/DependencyContextWriter.cs

```diff
@@ -171,9 +171,9 @@ private void WritePortableTarget(string key, IReadOnlyList<RuntimeLibrary> runti
                     Debug.Assert(compilationLibrary.RuntimeStoreManifestName == null);
                 }
 
-                Library library = (Library)compilationLibrary ?? (Library)runtimeLibrary;
+                Library? library = (Library?)compilationLibrary ?? runtimeLibrary;
 
-                WritePortableTargetLibrary(library.Name + DependencyContextStrings.VersionSeparator + library.Version,
+                WritePortableTargetLibrary(library?.Name + DependencyContextStrings.VersionSeparator + library?.Version,
```

Comment:
Previously this would throw a NullReferenceException if `library` was `null`. Now this will append garbage into the string.

We shouldn't change behavior here.